### PR TITLE
Update cluterctl version in e2e to latest patch

### PR DESCRIPTION
**Pull request by geetikabatra**
Update the clusterctl version used by upgrade e2e tests to a newer
patch version. The older patch versions might cause e2e tests to fail.

Signed-off-by: geetikab@vmware.com <geetikab@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: test/e2e/config/vsphere-ci.yaml [hunk 1]
```diff
@@ -165,8 +165,8 @@ variables:
   VSPHERE_STORAGE_POLICY: "Cluster API vSphere Storage Policy"
   VSPHERE_NETWORK: "sddc-cgw-network-6"
   VSPHERE_TEMPLATE: "ubuntu-1804-kube-v1.19.1"
-  INIT_WITH_BINARY_V1ALPHA3: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.23/clusterctl-{OS}-{ARCH}"
-  INIT_WITH_BINARY_V1ALPHA4: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.4/clusterctl-{OS}-{ARCH}"
+  INIT_WITH_BINARY_V1ALPHA3: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.4.7/clusterctl-{OS}-{ARCH}"
```
Comment (srm09) [inline test/e2e/config/vsphere-ci.yaml:168]: We should be using the `clusterctl` from the `v0.3.x` release line for this. This change is either not needed, or needs to be moved over to `v0.3.25` instead.

**Comment by srm09** [comment]
cc: @sdubey

**Comment by geetikabatra** [comment]
@srm09 Done!

**Pull request merged**